services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceHelixAdminWithAcl.java

```diff
@@ -0,0 +1,215 @@
+    // Setup authorizer with some default permissions
+    authorizerService = new MockVeniceAuthorizer();
+
+    // Add some default ACLs for testing
+    AclBinding defaultAcl = new AclBinding(new Resource(STORE_NAME));
+    defaultAcl.addAceEntry(new AceEntry(new Principal("admin"), Method.Read, Permission.ALLOW));
+    defaultAcl.addAceEntry(new AceEntry(new Principal("admin"), Method.Write, Permission.ALLOW));
+    authorizerService.setAcls(defaultAcl);
+
+    // Setup store with proper configuration
+    store = TestUtils.createTestStore(STORE_NAME, "owner", System.currentTimeMillis());
+    store.setMigrating(false);
+    store.setSystemStores(new HashMap<>());
+
+    // Create a mock VeniceHelixAdmin
+    veniceAdminMock = mock(VeniceHelixAdmin.class);
+
+    // Set up the mock to use our test authorizer service
+    when(veniceAdminMock.getAuthorizerService()).thenReturn(Optional.of(authorizerService));
+
+    // Set up the real method call for cleanupAclsForStore
+    doCallRealMethod().when(veniceAdminMock).cleanupAclsForStore(any(Store.class), anyString(), anyString());
+
+    // Reset counters before each test
+    authorizerService.setAclsCounter = 0;
+    authorizerService.clearAclCounter = 0;
+  }
+
+  @Test
+  public void testCleanupAclsForStore() {
+    // Setup test data with multiple ACEs
+    AclBinding aclBinding = new AclBinding(new Resource(STORE_NAME));
+    aclBinding.addAceEntry(new AceEntry(new Principal("user:test"), Method.Read, Permission.ALLOW));
+    aclBinding.addAceEntry(new AceEntry(new Principal("group:admins"), Method.Write, Permission.ALLOW));
+    authorizerService.setAcls(aclBinding);
+
+    // Verify initial state
+    AclBinding initialAcl = authorizerService.describeAcls(new Resource(STORE_NAME));
+    assertEquals(initialAcl.countAceEntries(), 2, "Should have 2 ACEs initially");
+
+    veniceAdminMock.cleanupAclsForStore(store, STORE_NAME, CLUSTER_NAME);
+
+    // Verify the authorizer service was called to clear ACLs for the main store
+    assertEquals(authorizerService.clearAclCounter, 1, "clearAcls should be called once");
+
+    // Verify the ACLs were actually cleared
+    AclBinding clearedAcl = authorizerService.describeAcls(new Resource(STORE_NAME));
+    assertEquals(clearedAcl.countAceEntries(), 0, "ACLs should be cleared for the main store");
+  }
+
+  @Test
+  public void testCleanupAclsForStoreWhenStoreIsNull() {
+    // Reset counter
+    authorizerService.clearAclCounter = 0;
+
+    // Call with null store - should log a warning but not throw
+    veniceAdminMock.cleanupAclsForStore(null, STORE_NAME, CLUSTER_NAME);
+
+    // Verify no interactions with authorizer service
+    assertEquals(authorizerService.clearAclCounter, 0, "clearAcls should not be called for null store");
+  }
+
+  @Test
+  public void testCleanupAclsForStoreWhenStoreIsMigrating() {
+    // Mark store as migrating
+    store.setMigrating(true);
+
+    // Reset counter
+    authorizerService.clearAclCounter = 0;
```

Comment:
```suggestion
  public void testCleanupAclsForStoreWhenStoreIsNull() {
    // Call with null store - should log a warning but not throw
    veniceAdminMock.cleanupAclsForStore(null, STORE_NAME, CLUSTER_NAME);

    // Verify no interactions with authorizer service
    assertEquals(authorizerService.clearAclCounter, 0, "clearAcls should not be called for null store");
  }

  @Test
  public void testCleanupAclsForStoreWhenStoreIsMigrating() {
    // Mark store as migrating
    store.setMigrating(true);
```
super nit: the counter should already be cleared in the `setup()`, right?

Reply:
done